File: src/gestortaller/Buseta.java
package gestortaller;
import gestorviajes.*;
public class Buseta extends Terrestre {
	private final static int NUM_SILLAS_ECONOMICAS = 6;
	private final static int NUM_SILLAS_EJECUTIVAS = 4;
	public Buseta(String nombre, Transportadora transportadora) {
		super(nombre, transportadora);
		this.setSILLASECONOMICAS(new Silla[NUM_SILLAS_ECONOMICAS]);
		this.setSILLASEJECUTIVAS(new Silla[NUM_SILLAS_EJECUTIVAS]);
		Ubicacion ubicacion;

		for (int numPosicion = 0; numPosicion < NUM_SILLAS_EJECUTIVAS; numPosicion++) {
			if (numPosicion % 4 == 0 || numPosicion % 4 == 3) {
				ubicacion = Ubicacion.VENTANA;
			} else {
				ubicacion = Ubicacion.PASILLO;
			}

			this.getSILLASEJECUTIVAS()[numPosicion] = new Silla(Clase.EJECUTIVA, numPosicion, ubicacion);
		}

		for (int numPosicion = 0; numPosicion < NUM_SILLAS_ECONOMICAS; numPosicion++) {
			if (numPosicion % 6 == 0 || numPosicion % 6 == 5) {
				ubicacion = Ubicacion.VENTANA;
			} else if (numPosicion % 6 == 1 || numPosicion % 6 == 4) {
				ubicacion = Ubicacion.CENTRAL;
			} else {
				ubicacion = Ubicacion.PASILLO;
			}
			this.getSILLASECONOMICAS()[numPosicion] = new Silla(Clase.EJECUTIVA, numPosicion, ubicacion);
		}
	}

	public static int getNumSillasEconomicas() {
		return NUM_SILLAS_ECONOMICAS;
	}

	public static int getNumSillasEjecutivas() {
		return NUM_SILLAS_EJECUTIVAS;
	}

	public String Calcular_Sillas_Ocupadas() {
		int cont = 0;
		for (Silla i : this.getSILLASECONOMICAS()) {
			if (i.isEstado()) {
				cont += 1;
			}
		}
		for (Silla j : this.getSILLASEJECUTIVAS()) {
			if (j.isEstado()) {
				cont += 1;
			}
		}
		return "Sillas ocupadas en la buseta"+cont;
	}
	public double Calcular_Consumo_Gasolina(double distancia_en_km) {
		double consumido;
		consumido = this.getGastoGasolina() * distancia_en_km;
		return consumido;
	}
}

File: practica02/Practica2VALE/Practica02VALE/SRC/gestorAplicacion/parada/Buseta.py

import math
from gestorAplicacion.parada.Terrestre import Terrestre
from gestorAplicacion.parada.Silla import Silla
from gestorAplicacion.parada.Clase import Clase
from gestorAplicacion.parada.Ubicacion import Ubicacion


class Buseta(Terrestre):

    _NUM_SILLAS_ECONOMICAS = 12
    _NUM_SILLAS_EJECUTIVAS = 4

    def __init__(self, nombre, transportadora):
        super().__init__(nombre, transportadora)
        
        for numPosicion in range(0, Buseta._NUM_SILLAS_EJECUTIVAS):
            if numPosicion % 4 == 0 or numPosicion % 4 == 3:
                ubicacion = Ubicacion.VENTANA
            else:
                ubicacion = Ubicacion.PASILLO

            self.getSILLASEJECUTIVAS().append(Silla(Clase.EJECUTIVA, numPosicion, ubicacion))

     
        for numPosicion in range(0, Buseta._NUM_SILLAS_ECONOMICAS):
            if numPosicion % 12 == 0 or numPosicion % 12 == 11:
                ubicacion = Ubicacion.VENTANA
            elif math.fmod(numPosicion, 12) == 1 or math.fmod(numPosicion, 12) == 4:
                ubicacion = Ubicacion.CENTRAL
            else:
                ubicacion = Ubicacion.PASILLO

            self.getSILLASECONOMICAS().append(Silla(Clase.ECONOMICA, numPosicion, ubicacion))

    @staticmethod
    def getNumSillasEconomicas():
        return Buseta._NUM_SILLAS_ECONOMICAS

    @staticmethod
    def getNumSillasEjecutivas():
        return Buseta._NUM_SILLAS_EJECUTIVAS


    def Calcular_Sillas_Ocupadas(self):
        cont = 0
        for i in self.getSILLASECONOMICAS():
            if i.isEstado():
                cont += 1
        for j in self.getSILLASEJECUTIVAS():
            if j.isEstado():
                cont += 1
        return cont


#consumo de gasolina 

    def Calcular_Consumo_Gasolina(self, distancia_en_km):
        consumido = self.getGastoGasolina() * distancia_en_km
        return consumido
